service/common/util/Message.kt

```diff
@@ -265,6 +265,9 @@ abstract class Message(codePrefix: String, codeNumber: Int, messagePrefix: Strin
                 Visualiser(1, "Unexpected error occurred in the graph visualiser.")
             val FULLY_EXPLAINED =
                 Visualiser(2, "This concept has been fully explained.")
+            val EXPLAIN_NOT_ENABLED =
+                Visualiser(3, "Explanations are not enabled. You must toggle 'explain' before " +
```

Comment:
To be extremely precise here, the user should have toggled 'explain` before _running the query_. The user's natural response to this warning in its current form would be to just click 'explain', try explaining again, and they'll end up either seeing the same warning, or getting the same error that they used to see before this PR.

We should re-test and see which of the above is true. If they end up getting the same error, we may need to rethink the approach. If they end up getting the friendly warning, we just need to make the warning both friendly and fully correct.

Reply:
Very good catch! This led me to another insight: we use `transactionState.transaction` in `GraphArea` and `GraphBuilder` with the assumption that this transaction is referencing the transaction for the graph. However, this assumption is faulty - we may well run another transaction elsewhere and come back to the previous graph.

To fix this, we've introduced two new properties of `GraphBuilder`:
```
    private val txCode = transactionState.transaction.hashCode()
    val transaction: TypeDBTransaction?
        get() = if (transactionState.transaction?.hashCode() == txCode) transactionState.transaction else null
```
Whenever we attempt to use the transaction, we check whether it's the same one we created this graph with. If not, `transaction` is `null`.